### PR TITLE
Fix uncaught errors in Desktop's Discover flow

### LINKED ISSUE #20202: Discover Desktops step throws an uncaught error
Once https://github.com/gravitational/teleport/issues/20148 is solved, we run into another error after clicking "Next" in the screen below:

Screenshot 2023-01-13 at 09 48 39 (2)

Initially the Discover Desktops screen loads:

Screenshot 2023-01-13 at 09 48 57 (2)

However after a few seconds, we get an uncaught error:

Screenshot 2023-01-13 at 09 49 04 (2)

### PR DESCRIPTION
**Pull request by ibeckermayer**
#### DiscoverDesktops.tsx
This won't happen in most cases, but in a case where a user defines a static host during the discovery flow, the `'teleport.dev/os'` and `'teleport.dev/os_version'` labels won't necessarily be defined, thus causing an error. 

#### `PingTeleportContext.tsx`
Instead of just checking for `!ctx.active` before calling `ctx.start`, we are instead checking for both `!ctx.active && !ctx.result`. (This logic has been moved from `usePingTeleport`'s `useEffect` to `ctx.start` itself). This way, [when we stop the `ctx`'s poll upon getting the `result` by setting `ctx.active` back to `false`](https://github.com/gravitational/teleport/blob/3ebcec2c9712a7d65c4dedf1c963d7dffa30ee61/web/packages/teleport/src/Discover/Shared/PingTeleportContext.tsx#L99) in `StartTeleportTerminalAnimation.tsx`, we don't also restart polling when `usePingTeleport` is called again in `DiscoverDesktops.tsx`

#### The rest
Clarifying comments

Fixes https://github.com/gravitational/teleport/issues/20202

### CODE REVIEW

**Review by kimlisa — APPROVED** [review]

**Review by kimlisa — APPROVED** [review]

**Review by ryanclark — APPROVED** [review]
❤️

File: web/packages/teleport/src/Discover/Shared/PingTeleportContext.tsx [hunk 1]
```diff
@@ -74,13 +78,24 @@ export function PingTeleportProvider<T>(props: {
...
 
-  const start = useCallback((joinToken: JoinToken) => {
-    setJoinToken(joinToken);
-    setActive(true);
-  }, []);
+  // start is called by usePingTeleport. It begins polling if polling is not
+  // yet active AND we haven't yet found a result.
+  const start = useCallback(
+    (joinToken: JoinToken, alternateSearchTerm?: string) => {
+      if (!active && !result) {
```
[the remark this answers is not in the document]
Comment (ibeckermayer) [reply web/packages/teleport/src/Discover/Shared/PingTeleportContext.tsx:85]: Agreed, `start` should just mean start

**Comment by ibeckermayer** [comment]
@kimlisa @ryanclark PTAL at this polishing commit https://github.com/gravitational/teleport/pull/21343/commits/c82e5fe49e73605bdd5412c0d5e3939822c7f7ad

**Review by kimlisa — APPROVED** [review]
lgtm after addressing @ryanclark comments

thanks for adding the clarifying comments, i really appreciate them

**Pull request merged**